config.c

```diff
@@ -13,14 +13,33 @@
 #include "criteria.h"
 #include "types.h"
 
+static struct mako_surface_config *create_surface(struct mako_config *config) {
+	struct mako_surface_config *ret = calloc(1, sizeof(*ret));
+	if (!ret) {
+		return NULL;
+	}
+
+	wl_list_insert(config->surfaces.prev, &ret->link);
+
+	ret->output = strdup("");
+	ret->layer = ZWLR_LAYER_SHELL_V1_LAYER_TOP;
+	ret->max_visible = 5;
+
+	ret->anchor =
+		ZWLR_LAYER_SURFACE_V1_ANCHOR_TOP | ZWLR_LAYER_SURFACE_V1_ANCHOR_RIGHT;
+
+	return ret;
+}
 
 static int32_t max(int32_t a, int32_t b) {
 	return (a > b) ? a : b;
 }
 
 void init_default_config(struct mako_config *config) {
 	wl_list_init(&config->criteria);
+	wl_list_init(&config->surfaces);
 	struct mako_criteria *new_criteria = create_criteria(config);
+	struct mako_surface_config *new_surface = create_surface(config);
```

Comment:
`create_surface` returns NULL if allocation fails but this isn't being handled here. We already do a mixed job of dealing with allocation failures in reasonable ways, so I'm not sure how careful we want to be here. I think it would be reasonable to bail all the way out if we fail this early.

Reply:
I'd rather not right now. The line directly above it (`create_criteria`) has the exact same issue.

Fixing that in a sane way requires changing the return type of the function. While I don't mind doing it, I'd rather outsource it into a different PR based on this, which also incorporates valgrind memcheck and clang scan-build based changes.